ecl/hql/hqlexpr.cpp

```diff
@@ -11510,6 +11528,27 @@ extern IHqlExpression *createDatasetF(node_operator op, ...)
     return createDataset(op, children);
 }
 
+IHqlExpression *createDataset(node_operator op, const std::initializer_list<IHqlExpression *> &operands)
+{
+    HqlExprArray args;
+    for (auto & cur : operands)
```

Comment:
No call to ensure() here?

Reply:
No - I also debated this pone.  It currently isn't because some could be commas and therefore exapanded, and some could be null.